biz.aQute.bnd.gradle/src/aQute/bnd/gradle/Bndrun.groovy

```diff
@@ -144,7 +145,21 @@ public class Bndrun extends DefaultTask {
         throw new GradleException("${run.getPropertiesFile()} workspace errors")
       }
 
-      worker(run)
+      def parent = run.parent;
+      try {
+        if (parent instanceof Processor) {
+          Properties gradleProperties = new PropertiesWrapper()
+          gradleProperties.put('task', this)
+          gradleProperties.put('project', project)
+          Processor processor = new ProcessorWrapper(gradleProperties)
+          processor.setParent(parent)
+          run.setParent(processor)
+        }
+
+        worker(run)
+      } finally {
```

Comment:
Not sure I see the utility in restoring the parent just before the run is closed. I would just avoid this.

Reply:
You may be right here since this may have been a side effect of earlier failed attempts. I'll try without this restoration.